### PR TITLE
fix: rework stacking flow, closes #335

### LINKED ISSUE #335: Send user back to home screen directly with clear activity upon initiation of Stacking
As soon as the user enters their password correctly to initiate Stacking, they should be sent directly back to the home screen where they can see that Stacking has been initiated in the activity stream instead of a blocking modal:

Screen Shot 2020-11-12 at 14 18 30

The relevant transaction should be shown as "Initiating Stacking..." instead of "Sending", which is misleading and a bit scary:

Screen Shot 2020-11-12 at 14 21 35

If initiation succeeds or fails, the transaction should then be shown as "Stacking initiated successfully" or "Stacking initiated failed" instead of "Contract call", which is generic and unclear: 

Screen Shot 2020-11-13 at 12 40 59

cc @jasperjansz for input in case he has a better idea

### PR DESCRIPTION
**Pull request by kyranjamie**
> [Download the latest builds [4.0.0-beta.8]](https://github.com/blockstack/stacks-wallet/actions/runs/393588694).

image

### CODE REVIEW

**Comment by markmhx** [comment]
Partial Stacking works well for me:

image

Only small confusing thing is seeing slightly less STX locked than I entered (85k), presumably because it was less fees. But that'll get addressed with https://github.com/blockstack/stacks-wallet/issues/279.

File: app/pages/stacking/components/stacking-info-card.tsx [hunk 1]
```diff
@@ -44,7 +48,7 @@ export const StackingInfoCard: FC<StackingInfoCardProps> = props => {
               Cycles
             </Text>
             <ExplainerTooltip>
-              One cycle lasts 1000 blocks on the bitcoin blockchain
+              One cycle lasts 120 blocks on the Bitcoin blockchain
```
Comment (markmhx) [inline app/pages/stacking/components/stacking-info-card.tsx:51]: This isn't a variable provided by the API is it?

**Review by markmhx — APPROVED** [review]

**Comment by kyranjamie** [comment]
Unless I'm mistaken, we can only estimate fees ahead of time, right? Is it guaranteed the actual fees will be the same?

Is it not equally confusing if you enter one number and then slightly more is transferred out? Are there other wallets that let you control whether the fee is taken from the sent amount or the remaining amount?

File: app/pages/stacking/components/stacking-info-card.tsx (hunk 1 above)
Comment (kyranjamie) [reply app/pages/stacking/components/stacking-info-card.tsx:51]: updated

File: app/utils/form/validate-decimals.ts [hunk 2]
```diff
@@ -0,0 +1,10 @@
+import BigNumber from 'bignumber.js';
+
+export function validateDecimalPrecision(allowedPrecision: number) {
```
Comment (hstove) [inline app/utils/form/validate-decimals.ts:3]: Maybe add a TSDoc to this - took me a minute to figure out what this does (although the function name is well descriptive)

File: app/utils/stacking/pox.spec.ts [hunk 3]
```diff
@@ -58,6 +58,7 @@ test.skip('making a lock-stx transaction', async () => {
   });
   await waitForTxConfirm(`0x${lockTxid}`);
   const stackerInfo = await client.getStackerInfo(address);
+  if ('error' in stackerInfo) return;
```
Comment (hstove) [inline app/utils/stacking/pox.spec.ts:61]: Since this is a test, maybe first add a test that there is no error, and if so, add an exception / return here?

**Review by hstove — APPROVED** [review]
Good work as always

File: app/utils/stacking/pox.spec.ts (hunk 3 above)
Comment (kyranjamie) [reply app/utils/stacking/pox.spec.ts]: This was just to satisfy types. Test is skipped and this code will be replaced by Ken's stacking lib soon.

File: app/pages/stacking/step/choose-amount.tsx [hunk 4]
```diff
@@ -0,0 +1,120 @@
...
+  balance: BigNumber;
+  step?: number;
+  minimumAmountToStack: number;
+  isComplete: boolean;
+  value: BigNumber | null;
+  onEdit(): void;
+  onComplete(amount: BigNumber): void;
+}
+
+const BigNumberFloorRound = BigNumber.clone({ ROUNDING_MODE: BigNumber.ROUND_FLOOR });
```
[the remark this answers is not in the document]
Comment (kyranjamie) [reply app/pages/stacking/step/choose-amount.tsx:28]: This let's me use BigNumber with the default rounding behaviour which is similar but not exactly the same as running a `Math.floor()`. I really hate the style of API they expose to do this, but generally prefer to do these types of operations within bignumber than `Math` methods.

**Comment by kyranjamie** [comment]
I'm going to merge this in, @timstackblock — you can test form `release/stacking`

**Pull request merged**